api/client/proxy.go

```diff
@@ -28,23 +29,34 @@ import (
 )
 
 // DialProxy creates a connection to a server via an HTTP Proxy.
-func DialProxy(ctx context.Context, proxyAddr, addr string) (net.Conn, error) {
-	return DialProxyWithDialer(ctx, proxyAddr, addr, &net.Dialer{})
+func DialProxy(ctx context.Context, proxyURL *url.URL, addr string) (net.Conn, error) {
+	return DialProxyWithDialer(ctx, proxyURL, addr, &net.Dialer{})
 }
 
 // DialProxyWithDialer creates a connection to a server via an HTTP Proxy using a specified dialer.
-func DialProxyWithDialer(ctx context.Context, proxyAddr, addr string, dialer ContextDialer) (net.Conn, error) {
-	conn, err := dialer.DialContext(ctx, "tcp", proxyAddr)
+func DialProxyWithDialer(ctx context.Context, proxyURL *url.URL, addr string, dialer ContextDialer) (net.Conn, error) {
+	conn, err := dialer.DialContext(ctx, "tcp", proxyURL.Host)
 	if err != nil {
-		log.Warnf("Unable to dial to proxy: %v: %v.", proxyAddr, err)
+		log.Warnf("Unable to dial to proxy: %v: %v.", proxyURL.Host, err)
 		return nil, trace.ConvertSystemError(err)
 	}
 
+	header := make(http.Header)
+	if proxyURL != nil && proxyURL.User != nil {
```

Comment:
If proxyURL == nil the code will panic in proxyURL.Host reference. Can we extract this check and move at the beginning of this function ?